CMSgov/mint-app
# Easi 3556/email selected solution pocs

# EASI-3556

## Changes and Description

- Introduced filter views to the possible operational solution table
- Created a new email template for when a solution is selected
    - Created a new database query to get the data (otherwise there would be 9 db queries at a minimum)
    - The email only sends for solutions that aren't custom (`isOther = false`)
    - The email only sends if there are defined POCs for that solution



## How to test this change
1. To simply validate the template, run `scripts/dev email`
2. Compare with [Figma](https://www.figma.com/file/8MfXzXzmKWv8vBjTasUIkD/MINT---Notifications?type=design&node-id=2921-759&mode=design&t=mOhoasphhUyCQcU6-0)

Otherwise,
1. Bring up the app and make a model plan.
2. Make a solution before updating plan basics performance_period_starts (this is the model start date)
3. Look at the email that comes (model start date should be excluded from the email.

4. Update the date in model basics, and choose a new solution
5. The new email should have the model start date


Filter Views
1. Select a solution that has a pre-defined filter view.
2. Make sure the email link uses that views language, and links correctly in MINT (4innovation (4i) is a good example, it should link to IDDOC)
3. Do the same with a solution without a filter view (RMADA is an example)
4. Make sure the email only links to the read view, and the language is correct




## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
